### PR TITLE
suite: recover and fail on subtest

### PR DESCRIPTION
**Pull request by anuragprafulla**

### CODE REVIEW

**Comment by dolmen** [comment]
Please spare the time of the maintainers by filling the description properly.

File: suite/suite.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func (suite *Suite) Run(name string, subtest func()) bool {
 	}()
 
 	return oldT.Run(name, func(t *testing.T) {
+		defer recoverAndFailOnPanic(t)
```
Comment (dolmen) [inline suite/suite.go:111]: This change needs a matching test case for future regression testing.

**Review by dolmen — CHANGES_REQUESTED** [review]

**Comment by xwjdsh** [comment]
@dolmen I submitted [a PR](https://github.com/stretchr/testify/pull/1293) long ago and was just reminded by a reply. I noticed this PR was duplicated with it, could you please take a look #1293?

**Comment by dolmen** [comment]
@xwjdsh There are changes in #1471 which are about to be merged and are also related to suite subtests. Could you add your review there and tell me if there will be something remaining to fix in #1455 and #1293?

**Comment by dolmen** [comment]
Closing because this change is included in #1471.

**Pull request closed without merging**